File: BMI.py
#!/usr/bin/env python
import rospy
from beginner_tutorials.msg import data
from beginner_tutorials.msg import info

def callback(data):
	rospy.loginfo("ID: %d" % (data.ID))
	rospy.loginfo("height: %f" % (data.height))
	rospy.loginfo("weight: %f" % (data.weight))
	msg = info()
	msg.ID = data.ID
	msg.BMI = (data.weight*10000)/(data.height*data.height)
	pub = rospy.Publisher('result', info, queue_size=10)
	pub.publish(msg)
	#rospy.loginfo("%d 's BMI is : %f" % (data.ID, data.BMI))

def listener():
    rospy.init_node('custom_listener', anonymous=True)
    rospy.Subscriber("data", data, callback)

    # spin() simply keeps python from exiting until this node is stopped
    rospy.spin()

if __name__ == '__main__':
    listener()

File: BMI.cpp
#include "ros/ros.h"
#include "beginner_tutorials/data.h"
#include "beginner_tutorials/info.h"


ros::Publisher *pub;

void chatterCallback(const beginner_tutorials::data::ConstPtr& msg)
{
  beginner_tutorials::info INFO;
  INFO.ID = msg->ID;
  INFO.BMI = (msg->weight)*10000/(msg->height*msg->height);

  ROS_INFO("ID : [%d]", INFO.ID);
  ROS_INFO("BMI : [%f]", INFO.BMI);
  pub->publish(INFO);
  
  
}


int main(int argc, char **argv)
{
	ros::init(argc, argv, "BMI");
	ros::NodeHandle n;

	ros::Subscriber sub = n.subscribe("data", 1000, chatterCallback);

	pub = new ros::Publisher(n.advertise<beginner_tutorials::info>("BMI_ID", 1000));

	//std_msgs::Float64 BMI;
	//BMI.data = Weight.data/(Height.data);


	
	//ros::Publisher BMI_pub = n.advertise<std_msgs::Float64>("BMI", 1000);
	//BMI_pub.publish(BMI);

	ros::spin();

	return 0;
}